util/carvelhelpers/package.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package carvelhelpers
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	yttui "github.com/vmware-tanzu/carvel-ytt/pkg/cmd/ui"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/files"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/workspace"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/workspace/datavalues"
+)
+
+const ConfigFilePermissions = 0o600
+
+// ProcessYTTPackage processes configuration directory with ytt tool
+// Implements similar functionality as `ytt -f <config-dir>`
+func ProcessYTTPackage(configDirs ...string) ([]byte, error) {
+	yttFiles, err := files.NewSortedFilesFromPaths(configDirs, files.SymlinkAllowOpts{})
+	if err != nil {
+		return nil, err
+	}
+
+	lib := workspace.NewRootLibrary(yttFiles)
+	libCtx := workspace.LibraryExecutionContext{Current: lib, Root: lib}
+	libExecFact := workspace.NewLibraryExecutionFactory(&NoopUI{}, workspace.TemplateLoaderOpts{})
+	loader := libExecFact.New(libCtx)
+
+	valuesDoc, libraryValueDoc, err := loader.Values([]*datavalues.Envelope{}, datavalues.NewNullSchema())
+	if err != nil {
+		return nil, err
+	}
+	result, err := loader.Eval(valuesDoc, libraryValueDoc, []*datavalues.SchemaEnvelope{})
+	if err != nil {
+		return nil, err
+	}
+	return result.DocSet.AsBytes()
+}
+
+// CarvelPackageProcessor processes a carvel package and returns a configuration YAML file
+func CarvelPackageProcessor(pkgDir string) ([]byte, error) {
+	// Each package contains `config` and `.imgpkg` directory
+	// `config` directory contains ytt files
+	// `.imgpkg` directory contains ImageLock configuration for ImageResolution
+	configDir := filepath.Join(pkgDir, "config")
+	file := []string{configDir}
+	bytes, err := ProcessYTTPackage(file...)
+	if err != nil {
+		return nil, errors.Wrap(err, "error while running ytt")
+	}
+
+	f, err := os.CreateTemp("", "ytt-processed")
```

Comment:
Is the variable `f` short for `file`? If so, is it the same type as `file` in line 51? If it is, I'd say, just use `file`. Single letter variables should only be for very obvious things, and I'm not so familiar with `os.CreateTemp` to know what `f` is supposed to be.

Oh, I see that `file` in line 51 is a slice of string. In that case, that line 51 variable should be plural, `files := []string{configDir}`. And that frees up `file` to then be used here in line 57.

Reply:
@codegold79  using files as variable will cause lint error 
**_importShadow: shadow of imported from 'github.com/vmware-tanzu/carvel-ytt/pkg/files' package 'files'_**

Reply:
Oh, I see. Carvel stole the name `files` by naming one of their packages that. How unfortunate. Well, looks like what you got is the best I can think of to do. Resolving.